### PR TITLE
Manage properly the oversampling situation

### PR DESCRIPTION
**Pull request by micheles**
See https://bugs.launchpad.net/oq-engine/+bug/1425030.
The tests are green: https://ci.openquake.org/job/zdevel_oq-risklib/322

### CODE REVIEW

File: openquake/commonlib/source.py [hunk 1]
```diff
@@ -394,6 +407,13 @@ def get_trt_id(self, col_id):
         raise KeyError('There is no TrtModel associated to the collection %d!'
                        % col_id)
 
+    def get_triples(self):
+        """
+        Yield triples (trt_id, idx, col_id) in order
```
Comment (ptormene) [inline openquake/commonlib/source.py]: Please rename col_id to col_idx also here

File: openquake/commonlib/source.py [hunk 2]
```diff
@@ -454,24 +474,20 @@ def get_rlzs_assoc(self, get_weight=lambda tm: tm.num_ruptures):
...
-            else:
-                tot_weight = sum(rlz.weight for rlz in assoc.realizations)
-                if tot_weight == 0:
-                    raise ValueError('All realizations have zero weight??')
-                elif tot_weight < 1:
-                    logging.warn('Some source models are not contributing, '
-                                 'weights are being rescaled')
-                for rlz in assoc.realizations:
-                    rlz.weight = rlz.weight / tot_weight
+        if num_samples:
```
Comment (ptormene) [inline openquake/commonlib/source.py]: Please check if there's actually any difference between the current situation and the old one

File: openquake/commonlib/source.py (hunk 1 above)
Comment (micheles) [inline openquake/commonlib/source.py]: Done

File: openquake/commonlib/source.py (hunk 2 above)
Comment (micheles) [inline openquake/commonlib/source.py]: There is no real difference. I am reverting to the original.

File: openquake/commonlib/calculators/classical.py [hunk 3]
```diff
@@ -22,6 +22,8 @@
 from functools import partial
 import numpy
 
+import numpy
```
Comment (ptormene) [inline openquake/commonlib/calculators/classical.py]: numpy was already imported

Comment (micheles) [inline openquake/commonlib/calculators/classical.py]: Ops, this was a merge error

File: openquake/commonlib/readinput.py [hunk 4]
```diff
@@ -414,11 +412,9 @@ def get_source_models(oqparam, source_model_lt, sitecol=None):
                 trt_model.gsims = gsim_lt.values[trt_model.trt]
         else:
             gsim_lt = logictree.DummyGsimLogicTree()
-        weight = (rlz.weight if oqparam.calculation_mode == 'event_based'
-                  else rlz.weight / num_samples)
+        weight = rlz.weight / num_samples
```
Comment (ptormene) [inline openquake/commonlib/readinput.py]: We might add a check in case num_samples is zero

**Comment by ptormene** [comment]
Apart from the minor comment, LGTM

File: openquake/commonlib/readinput.py (hunk 4 above)
Comment (micheles) [inline openquake/commonlib/readinput.py]: It cannot happen. num_samples is built by counting the source model realizations, so at minimum it is 1.

**Pull request merged**